.github/workflows/build-and-deploy.yml

```diff
@@ -1,19 +1,60 @@
 name: Build and deploy on push
 
 on:
+  workflow_dispatch:
   push:
     branches:
+      - development
       - master
+    tags:
+      - "*"
 
 jobs:
   docker:
+    name: Docker
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
     steps:
-      - uses: actions/checkout@v1
-      - name: Build Docker Image
-        uses: docker/build-push-action@v1
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set Docker Tag
+        id: tag
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            DOCKER_TAG="${GITHUB_REF:10}"
+          elif [[ $GITHUB_REF == refs/heads/development ]]; then
+            DOCKER_TAG="dev"
+          elif [[ $GITHUB_REF == refs/heads/master ]]; then
+            DOCKER_TAG="latest"
+          else
+            DOCKER_TAG="${GITHUB_REF:11}"
+          fi
+          echo ::set-output name=tag::${DOCKER_TAG}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: biospheere/promcord
-          tags: latest
\ No newline at end of file
+
+      - name: Build and Push Docker Image
+        if: github.ref == 'refs/heads/development' || github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/')
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: .
+          file: ./Dockerfile
+          platforms: linux/386,linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64/v8
```

Comment:
Do i need any changes here if we change the base image to 3.8.1-adoptopenjdk-16?

Reply:
As long as all of the platforms from the list in the workflow are provided by your base image, you should be good to go.
I would review the platforms lists on changing the base image.

Reply:
Alright. I'm going to change it in #96 to 3.8.1-adoptopenjdk-16 👍

Reply:
I had a look at the Dockerfile and wonderd if the openjdk image in line 6 is going to cause any issues?

Reply:
I can't say I fully understand why line 6 exists, the JDK is in the maven image. I'll run a few test builds

Reply:
Thats a multitstage docker build to reduce the image size